cmd/compose/top.go

```diff
@@ -63,30 +66,84 @@ func runTop(ctx context.Context, dockerCli command.Cli, backend api.Service, opt
 		return containers[i].Name < containers[j].Name
 	})
 
+	header, entries := collectTop(containers)
+	return topPrint(dockerCli.Out(), header, entries)
+}
+
+func collectTop(containers []api.ContainerProcSummary) (topHeader, []topEntries) {
+	// map column name to its header (should keep working if backend.Top returns
+	// varying columns for different containers)
+	header := topHeader{"SERVICE": 0, "#": 1}
+
+	// assume one process per container and grow if needed
+	entries := make([]topEntries, 0, len(containers))
+
 	for _, container := range containers {
-		_, _ = fmt.Fprintf(dockerCli.Out(), "%s\n", container.Name)
-		err := psPrinter(dockerCli.Out(), func(w io.Writer) {
-			for _, proc := range container.Processes {
-				info := []interface{}{}
-				for _, p := range proc {
-					info = append(info, p)
+		for _, proc := range container.Processes {
+			svc := container.Name
+			if tmp, ok := container.Labels[api.ServiceLabel]; ok {
```

Comment:
better introduce `ContainerProcSummary.Service` and `ContainerProcSummary.Replica` and initialize these in `composeService.Top` to limit code responsibility in cmd to formatting

Reply:
Done.